tests/test_parser.py

```diff
@@ -179,6 +195,39 @@ def test_include_role_parsing(
     assert include_role_4.has_loop(), "The third include role has a loop"
 
 
+@pytest.mark.parametrize(
+    "exclude_roles",
+    [
+        None,
+        ([]),
+        (["fake_role"]),
+        (["fake_role", "display_some_facts"]),
+    ],
+    ids=["none", "empty_list", "exclude_single_role", "exclude_multiple_roles"],
+)
+@pytest.mark.parametrize("grapher_cli", [["include_role.yml"]], indirect=True)
+def test_include_role_parsing_with_exclude_roles(
+    grapher_cli: PlaybookGrapherCLI, exclude_roles: list[str]
+) -> None:
+    """Test parsing of include_role
+    :param grapher_cli:
+    :param exclude_roles: flag to exclude certain roles
+    :return:
+    """
+    parser = PlaybookParser(
+        grapher_cli.options.playbooks[0],
+        include_role_tasks=True,
+        exclude_roles=exclude_roles,
+    )
+    playbook_node = parser.parse()
+    
+    # If exclude roles option is set then there should be no roles with the names identical to the option arguments
+    if exclude_roles is not None:
+        all_roles = get_all_roles([playbook_node])
+        role_names = list(map(lambda role_node: role_node.name, all_roles))
+        assert exclude_roles not in role_names
```

Comment:
You are comparing two lists here and this not what we want. For example, this returns true `["display_some_facts"] not in ["display_some_facts"]`.

You have to check one by one.